### PR TITLE
[msmpi] Upgrade to version 10.1

### PR DESCRIPTION
**Pull request by LilyWangL**
Related issue: #8393
There are no features of this port need to test.

### CODE REVIEW

**Comment by LilyWangL** [comment]
Build failed on CI with this error:
```
CMake Error at ports/msmpi/portfile.cmake:43 (message):
    The version of the installed MSMPI redistributable packages does not match the version to be installed
      Expected version: 10.1.12498
      Found version: 10.0.12498
    Please upgrade the installed version on your system.
    The appropriate installer for the expected version has been downloaded to:
      C:/vsts/_work/2/s/downloads/msmpisetup-10.1.12498.exe
```
This need to run 10.1.12498.exe manually. It will uninstall version 10.0.12498 of MPI and install new version of MPI.

**Review by NancyLi1013 — APPROVED** [review]

File: ports/msmpi/portfile.cmake [hunk 1]
```diff
@@ -4,20 +4,20 @@ if(VCPKG_CMAKE_SYSTEM_NAME)
     message(FATAL_ERROR "This port is only for building msmpi on Windows Desktop")
 endif()
 
-set(MSMPI_VERSION "10.0.12498")
+set(MSMPI_VERSION "10.1.12498")
```
Comment (ras0219-msft) [inline ports/msmpi/portfile.cmake]: Should this be 10.1.1?

Comment (LilyWangL) [reply ports/msmpi/portfile.cmake]: The version of Microsoft MPI SDK is 10.1.12498.16, so this is 10.1.12498. If change to 10.1.1, msmpi will build failed with this error:
```
The version of the installed MSMPI redistributable packages does not match the version to be installed
    Expected version: 10.1.1
    Found version: 10.1.12498
Please upgrade the installed version on your system.
```

**Comment by LilyWangL** [comment]
Closed as duplicate of #11365, #11365 has been merged.

**Pull request closed without merging**